eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Gateway/DoctrineDatabase.php

```diff
@@ -1160,4 +1174,36 @@ public function bulkRemoveTranslation($languageId, $actions)
         ;
         $query->execute();
     }
+
+    /**
+     * Cleanup obsolete lang_mask entries for the given Location.
+     *
+     * @param int $locationId
+     * @param int[] $newLanguageIds
+     * @param bool $alwaysAvailable
+     */
+    public function cleanupTranslationsAfterPublish($locationId, $newLanguageIds, $alwaysAvailable)
```

Comment:
Note that there's the method `cleanupAfterPublish`, however it relies on a single language, making it impossible to distinguish which ones got removed. I tried to reuse it, but it would lead to refactoring of SPI `publishUrlAliasForLocation` which is used in many places.